File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/operation/WhoOperation.java
package me.ycdev.android.lib.common.demo.service.operation;

import android.os.RemoteException;
import androidx.annotation.NonNull;

import java.util.concurrent.CountDownLatch;

import me.ycdev.android.lib.common.demo.service.IDemoService;
import me.ycdev.android.lib.common.ipc.IpcOperation;

public class WhoOperation implements IpcOperation<IDemoService> {
    private CountDownLatch mLatch;

    public WhoOperation setNotifier(CountDownLatch latch) {
        mLatch = latch;
        return this;
    }

    @Override
    public void execute(@NonNull IDemoService service) throws RemoteException {
        service.who();
        if (mLatch != null) {
            mLatch.countDown();
        }
    }

    @Override
    public String toString() {
        return "Operation[who]";
    }
}

File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/operation/WhoOperation.kt
package me.ycdev.android.lib.common.demo.service.operation

import android.os.RemoteException
import androidx.annotation.NonNull
import java.util.concurrent.CountDownLatch
import me.ycdev.android.lib.common.demo.service.IDemoService
import me.ycdev.android.lib.common.ipc.IpcOperation

class WhoOperation : IpcOperation<IDemoService> {
    private var mLatch: CountDownLatch? = null

    fun setNotifier(latch: CountDownLatch): WhoOperation {
        mLatch = latch
        return this
    }

    @Throws(RemoteException::class)
    override fun execute(@NonNull service: IDemoService) {
        service.who()
        if (mLatch != null) {
            mLatch!!.countDown()
        }
    }

    override fun toString(): String {
        return "Operation[who]"
    }
}
